server/engine/tests/test_missing_grf_detector.py

```diff
@@ -0,0 +1,27 @@
+import nimblephysics as nimble
+import unittest
+import os
+from inspect import getsourcefile
+import shutil
+from passes.missing_grf_detection import missing_grf_detection
+import tempfile
+
+TESTS_PATH = os.path.dirname(getsourcefile(lambda:0))
+TEST_DATA_PATH = os.path.join(TESTS_PATH, 'data')
+
+
+class TestMissingGRFDetection(unittest.TestCase):
+    def test_missing_grf_detection(self):
+        path = os.path.join(TEST_DATA_PATH, 'b3ds', 'falisse2017_small.b3d')
+        subject = nimble.biomechanics.SubjectOnDisk(path)
+        subject.loadAllFrames(doNotStandardizeForcePlateData=True)
+
+        # output_folder = os.path.join(TEST_DATA_PATH, 'temp_opensim_results')
+        # write_opensim_results(subject, output_folder)
+
+        num_passes = subject.getNumProcessingPasses()
+        missing_grf_detection(subject)
+        num_passes_after = subject.getNumProcessingPasses()
+
+        # This shouldn't add a pass
+        self.assertEqual(num_passes, num_passes_after)
```

Comment:
Would it make sense to add more unittests checking that individual missing frames were actually detected?

Reply:
This is the only one of your comments that I'm going to leave as a TODO in the code, because I think I'm not entirely sure how to do it. The GRF detector was originally developed in another repo, here: https://github.com/keenon/MissingGRFEstimator, and it works pretty well on the whole published AddBiomechanics dataset, but it's not obvious how to encapsulate that as a useful unit test (and not just a "make sure you label exactly these frames", which would just function as a change-detector and not necessarily a useful test)